### DISCUSSION TITLE
How to handle navbar "external" links without absolute paths?

### DISCUSSION THREAD

**Discussion by hasezoey** [facebook/docusaurus#9376] [Q&A] [answered]
Hello, i currently want to add a navbar link (in the top) for a subfolder that links to `typedoc` output (not generated via docusaurus), i currently have the following configuration:

```js
{
  href: 'typedoc/index.html',
  prependBaseUrlToHref: true, // even without this option
  label: 'Typedoc',
  position: 'right',
},
```

but sadly this gives me errors about broken links:

```txt
[ERROR] Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

It looks like some of the broken links we found appear in many pages of your site.
Maybe those broken links appear on all pages through your site layout?
We recommend that you check your theme configuration for such links (particularly, theme navbar and footer).
Frequent broken links are linking to:
- /mongodb-memory-server/typedoc/index.html

Exhaustive list of all broken links found:

- On source page path = /mongodb-memory-server/404.html:
   -> linking to /mongodb-memory-server/typedoc/index.html

- On source page path = /mongodb-memory-server/search:
   -> linking to /mongodb-memory-server/typedoc/index.html

- On source page path = /mongodb-memory-server/docs/api/classes/dry-mongo-binary:
   -> linking to /mongodb-memory-server/typedoc/index.html

- On source page path = /mongodb-memory-server/docs/api/classes/mongo-binary:
   -> linking to /mongodb-memory-server/typedoc/index.html

...
```

the build error can be worked-around by using `onBrokenLinks: 'warn',`, but i would prefer to error on actual broken link errors, it also causes some runtime problems, because react thinks that it is a actual route that exists and so gives a 404 until reloaded.

the href needs to be relative (and not absolute) because docusaurs may get build for different versions and so have different base paths (see docusaurus.config for details, custom versioning) or also run locally, how do i do this properly?

TL;DR: i want a navbar link that can redirect to `/<base-url>/customPath` (`/mongodb-memory-server/typedoc/index.html`), how do i do this properly?

Repository & branch in question: https://github.com/nodkz/mongodb-memory-server/tree/testTsDoc
(specific commit at question open time is `1eaeaddda3a64a82bcd7ae4c227905d702eb135e`)

**Comment by hasezoey** [comment]
for now i have gone with a custom component, but it would be great to know how to do this without a custom component as can be seen in https://github.com/nodkz/mongodb-memory-server/commit/48402d6448dadd7d2f4456a466a25db286873440

**Comment by Josh-Cena** [comment]
Try `to: "pathname://typedoc/index.html`? See https://docusaurus.io/docs/advanced/routing#escaping-from-spa-redirects

**Reply by hasezoey** [reply]
yes i can confirm that this works, but there is still some unwanted behavior, so i will likely stay with the custom component:
- it is marked as a external link (even though it is on the same website)
- it has `target="_blank"`, which will open a new tab instead of re-using the current one

**Reply by slorber** [reply]
This historical link component is a bit confusing, and not sure all of these works but you can try this:
- use `to` instead of `href`, I think it removes the external icon
- use `target: undefined` (or _self)
- use `data-noBrokenLinkCheck: true`

**Reply by hasezoey** [reply, marked as answer]
can confirm that the following works (not marked external, opens in same tab, no broken link error (even if it does not exist), actually opens and not a 404):

```js
{
  // triple "/"
  to: 'pathname:///typedoc/index.html',
  label: 'Typedoc',
  position: 'right',
  target: '_self',
},
```

i would appreciate those things to be documented as possible options in [Navbar: Navbar-link](https://docusaurus.io/docs/api/themes/configuration#navbar-link)

**Reply by slorber** [reply]
Great

If you have any doc improvement idea please submit a PR directly

`target="_self"` is kind of documented already:

<img width="978" alt="CleanShot 2023-10-08 at 18 39 20@2x" src="https://github.com/facebook/docusaurus/assets/749374/a4a6b570-4ac0-45c9-b5a7-dd4b4672c3de">

![CleanShot 2023-10-08 at 18 40 07@2x](https://github.com/facebook/docusaurus/assets/749374/ee295c90-5944-4ccf-b6c7-04689fb3a89d)


`pathname://` documented here: https://docusaurus.io/docs/advanced/routing#escaping-from-spa-redirects

Regarding `to` vs `href` I'm not even sure to understand what the historical differences are exactly so not sure how to document this 😅 we'll have to clean this up a bit for sure in the future.
